go/vt/vtgate/planbuilder/subquery.go

```diff
@@ -120,12 +119,22 @@ func (sq *subquery) PushSelect(expr *sqlparser.AliasedExpr, _ builder) (rc *resu
 	return rc, len(sq.resultColumns) - 1, nil
 }
 
-// PushOrderByNull satisfies the builder interface.
-func (sq *subquery) PushOrderByNull() {
+// MakeDistinct satisfies the builder interface.
+func (sq *subquery) MakeDistinct() error {
+	return errors.New("unsupported: distinct on cross-shard subquery")
 }
 
-// PushOrderByRand satisfies the builder interface.
-func (sq *subquery) PushOrderByRand() {
+// PushGroupBy satisfies the builder interface.
+func (sq *subquery) PushGroupBy(_ sqlparser.GroupBy) error {
+	return errors.New("unupported: group by on cross-shard subquery")
```

Comment:
```suggestion
	return errors.New("unsupported: group by on cross-shard subquery")
```

Reply:
Why didn't this fail a unit test?

Reply:
This code is unreachable because there are higher level checks that catch this case. But it can get called in the future as things evolve.

It's something one has to keep in mind while making changes: review code coverage of tests and make sure reachable code is reached.